opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -358,6 +371,11 @@ static int accelerator_cuda_memcpy(int dest_dev_id, int src_dev_id, void *dest,
 {
     CUresult result;
 
+    result = opal_accelerator_cuda_delayed_init();
+    if (0 != result) {
+        return result;
+    }
```

Comment:
I think it might be worth wrapping some of the error checks in `OPAL_UNLIKELY()`. I did this locally for a few of these and saw a small performance boost in some OMB benchmarks.

```suggestion
    result = opal_accelerator_cuda_delayed_init();
    if (OPAL_UNLIKELY(0 != result)) {
        return result;
    }
```

Reply:
I wrapped the ones in this PR, I'll maybe add a separate PR wrapping the other error paths